lib/src/react_controller/ctrlThread.cpp

```diff
@@ -60,6 +69,26 @@ CtrlThread::CtrlThread(const std::string& _name, const std::string& _limb, bool
     }
 }
 
+void CtrlThread::initializeApp(bool _verbosity)
+{
+    app=new Ipopt::IpoptApplication;
+    app->Options()->SetNumericValue("tol",tol);
+    app->Options()->SetNumericValue("constr_viol_tol",1e-6);
+    // app->Options()->SetIntegerValue("acceptable_iter",0);
+    app->Options()->SetStringValue ("mu_strategy","adaptive");
+    if (is_debug == false) {
+        app->Options()->SetStringValue ("linear_solver", "ma57");
+    }
+    app->Options()->SetIntegerValue("max_iter",std::numeric_limits<int>::max());
+    app->Options()->SetNumericValue("max_cpu_time", 0.95 * dT);
+    // app->Options()->SetStringValue ("nlp_scaling_method","gradient-based");
+    app->Options()->SetStringValue ("hessian_approximation","limited-memory");
+    // app->Options()->SetStringValue ("derivative_test",verbosity?"first-order":"none");
+    app->Options()->SetStringValue ("derivative_test","none");
+    app->Options()->SetIntegerValue("print_level",_verbosity & !is_debug?5:0);
```

Comment:
What codacy said is correct, please clarify calculation precedence for '+' and '?'.